src/framework/theme/components/select/option-group.component.ts

```diff
@@ -26,13 +40,49 @@ export class NbOptionGroupComponent {
   }
   set disabled(value: boolean) {
     this._disabled = convertToBoolProperty(value);
+
+    if (this.options) {
+      this.updateOptionsDisabledState();
+    }
   }
   protected _disabled: boolean = false;
 
   @HostBinding('attr.disabled')
   get disabledAttribute(): '' | null {
     return this.disabled ? '' : null;
   }
+
+  @ContentChildren(NbOptionComponent, { descendants: true }) options: QueryList<NbOptionComponent<any>>;
+
+  ngAfterContentInit() {
+    if (this.options.length) {
+      this.asyncUpdateOptionsDisabledState();
+    }
+
+    this.options.changes
+      .pipe(takeWhile(() => this.alive))
```

Comment:
Maybe, let's start using `takeUntil(this.destroyed$)` 😄 in new components?

Reply:
I think we'll refactor components all together later and before that moment we should stick with a current agreed style. Consistency, you know. 🙂